tests/test_frame.py

```diff
@@ -655,21 +660,23 @@ def test_types_groupby_agg() -> None:
         assert_type(df.groupby("col1").agg(["min", "max"]), pd.DataFrame), pd.DataFrame
     )
     check(assert_type(df.groupby("col1").agg([min, max]), pd.DataFrame), pd.DataFrame)
+    agg_dict1: dict[Hashable, str] = {"col2": "min", "col3": "max", 0: "sum"}
```

Comment:
Can you add a comment above this line indicating why `Hashable` has to be used?  Something along the idea that the type-checkers may or may not infer the type of the `dict` correctly.

Would `agg_dict1: dict[Union[int, str], str] = {"col2": "min", "col3": "max", 0: "sum"}` work?  (which might be more natural to a user than using `Hashable`)